sdk/storage/Azure.Storage.Queues/tests/QueueClientTests.cs

```diff
@@ -1977,5 +1979,68 @@ public void CanMockClientConstructors()
             mock = new Mock<QueueClient>(new Uri("https://test/test"), new AzureSasCredential("foo"), new QueueClientOptions()).Object;
             mock = new Mock<QueueClient>(new Uri("https://test/test"), mockTokenCredential, new QueueClientOptions()).Object;
         }
+
+        [RecordedTest]
+        [TestCase("")]
+        [TestCase("u")]
+        [TestCase("au")]
+        [TestCase("rap")]
+        [TestCase("raup")]
+        public async Task QueueClientPolicyPermissions_checkIfSetCorrectly(string permissions)
+        {
+            await using DisposingQueue test = await GetTestQueueAsync();
+
+            QueueSignedIdentifier[] signedIdentifiers = new[]
+            {
+                new QueueSignedIdentifier
+                {
+                    Id = GetNewString(),
+                    AccessPolicy =
+                        new QueueAccessPolicy
+                        {
+                            StartsOn =  Recording.UtcNow.AddHours(-1),
+                            ExpiresOn =  Recording.UtcNow.AddHours(1),
+                            Permissions = permissions
+                        }
+                }
+            };
+            Response setResult = await test.Queue.SetAccessPolicyAsync(signedIdentifiers);
+
+            Response<IEnumerable<Models.QueueSignedIdentifier>> getResult = await test.Queue.GetAccessPolicyAsync();
+            Models.QueueSignedIdentifier acl = getResult.Value.First();
+
+            string actualPermissionsStr = acl.AccessPolicy.Permissions;
+            QueueAccessPolicyPermissions? actualPermissionsEnum = acl.AccessPolicy.QueueAccessPolicyPermissions;
+
+            string expectedPermissionsStr = null;
+            QueueAccessPolicyPermissions? expectedPermissionsEnum = null;
+
+            switch (permissions)
```

Comment:
Instead of having this switch case, would it possible to add one more parameters to the TestCase(s)

so it would be more like
```
[TestCase("", null)]
[TestCase("u", QueueAccessPolicyPermissions.Update)]
[TestCase("au",QueueAccessPolicyPermissions.Add | QueueAccessPolicyPermissions.Update )]
[TestCase("rap",  QueueAccessPolicyPermissions.Read | QueueAccessPolicyPermissions.Add | QueueAccessPolicyPermissions.Process)]
[TestCase("raup", QueueAccessPolicyPermissions.All)]
public async Task QueueClientPolicyPermissions_checkIfSetCorrectly(string permissions, QueueAccessPolicyPermissions? expectedPermissionEnum)
```

And also since `expectedPermissionsStr` will be the same value as `permissions` you can make that equal unless `permissions` is and empty string

Reply:
Yeah I was thinking of adding the permissionsEnum as a parameter instead of having the switch cases, but there were exceptions to the case such as the null/"". 

But yeah I can prob handle the edge cases individually